lib/facility.rb

```diff
@@ -1,14 +1,68 @@
+require 'pry'
+require './lib/facility'
+require './lib/vehicle'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
+    @services = facility_details[:services] || []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
-
-  def add_services(service)
+  
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)    
+    return nil unless @services.include?('Vehicle Registration')    
+
+    unless @registered_vehicles.include?(vehicle)
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Date.today  
+
+      #require 'pry'; binding.pry
+
+      if vehicle.antique?
+        @collected_fees += 25
```

Comment:
Consider collecting fees in a small helper method as well, to better promote SRP